common/src/main/java/com/wynntils/core/webapi/profiles/MapProfile.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.webapi.profiles;
+
+import com.wynntils.mc.utils.McUtils;
+import net.minecraft.client.renderer.texture.DynamicTexture;
+import net.minecraft.resources.ResourceLocation;
+
+public class MapProfile {
+
+    DynamicTexture texture;
+    ResourceLocation mapResource;
+
+    int x1;
+    int z1;
+    int x2;
+    int z2;
+
+    int textureWidth;
+    int textureHeight;
+
+    public MapProfile(DynamicTexture texture, int x1, int z1, int x2, int z2, int textureWidth, int textureHeight) {
+        this.texture = texture;
+        this.x1 = x1;
+        this.z1 = z1;
+        this.x2 = x2;
+        this.z2 = z2;
+        this.textureWidth = textureWidth;
+        this.textureHeight = textureHeight;
+
+        // Remove this if we ever have non 1 to 1 maps
+        assert x2 - x1 == textureWidth;
```

Comment:
We can calculate width and height from the input coordinates. 

However, we need to validate that this information matches the actual texture. So you should check the calculated width/height (from x12/z12) with `texture.getPixels().getWidth()` etc. And if it fails, throw an exception (a runtime exception is fine, you don't have to declare it).

Reply:
How does this differ from what occurs right now?

Reply:
1) You need to pass in the height to the constructur, even though it can be calculated. Instead you have an assert to verify that the calculation is correct.

2) You need to check the actual texture dimensions.

Reply:
Okay, I see now that you send in the texture size. But that's putting the responsibility on the caller. This should be done by this class instead.

Reply:
Will do.